core/src/test/scala/ConsulOpTests.scala

```diff
@@ -67,4 +67,45 @@ class ConsulOpTests extends FlatSpec with Matchers with TypeCheckedTripleEquals
     } yield ()
     interp.run(healthCheckJson[HealthStatus]("foo")).run should equal(\/.right(List()))
   }
+
+  "catalogServices" should "return a vector of registered services" in {
+    val interp = for {
+      _ <- I.expectU[Option[String]] {
+        case ConsulOp.GetCatalogServices => now(Some("""{"consul":[],"service-name":["tag1","tag2"]}"""))
+      }
+    } yield ()
+    interp.run(getJson[Json]("service-name")).run should equal(\/.right(Some(List("tag1","tag2"))))
```

Comment:
You don't want to be calling `getJson` here as that's top level api that calls consul's ket value api and decodes the result as json. 

see here: https://github.com/Verizon/helm/blob/master/core/src/main/scala/HelmOp.scala#L33

What you probably want to follow this pattern with your code. So you'd have

```scala
def getService(service: String): ConsulOpF[Option[String]]
```

and then another top level method that calls `getService` and decodes the json with the given deocder.

```scala
def getServiceJson[A:DecodeJson](service: String): ConsulOpF[Err \/ Option[A]]
```